### PR TITLE
Added support for sender id configuration per country

### PR DESCRIPTION
**Pull request by sbryzak**
This PR modifies registration service to allow sender IDs to be configured for individual country codes.

Fixes https://issues.redhat.com/browse/SANDBOX-323

Tests have been conducted manually as they required live Twilio keys.

image

### CODE REVIEW

File: pkg/verification/sender/amazon_sns_sender.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func NewAmazonSNSSender(cfg AWSSenderConfiguration) NotificationSender {
 	}
 }
 
-func (s *AmazonSNSSender) SendNotification(_ *gin.Context, content, phoneNumber string) error {
+func (s *AmazonSNSSender) SendNotification(_ *gin.Context, content, phoneNumber, _ string) error {
```
Comment (alexeykazakov) [inline pkg/verification/sender/amazon_sns_sender.go:30]: Maybe add a TODO note to implement it for AWS?

**Review by alexeykazakov — COMMENTED** [review]
The code looks fine, however I don't see any actual unit test for this code. Am I missing something?

File: pkg/verification/sender/amazon_sns_sender.go (hunk 1 above)
Comment (sbryzak) [reply pkg/verification/sender/amazon_sns_sender.go:30]: added TODO

**Comment by sbryzak** [comment]
Unit test added

File: pkg/verification/sender/twilio_sender_test.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			{
+				SenderID:     "RED HAT",
+				CountryCodes: []string{"44"},
+			},
+		},
+	}
+
+	sender := sender2.NewTwilioSender(cfg, httpClient)
+
+	err := sender.SendNotification(ctx, "Test Message", "+440000000000", "44")
```
Comment (alexeykazakov) [inline pkg/verification/sender/twilio_sender_test.go:78]: Can we also add a test which would send to a number which doesn't have the country code representation in the config? Basically another notification to any country code but "44". To check it doesn't use the sender ID and not just fail.

**Review by alexeykazakov — COMMENTED** [review]
Thank you for the test! Looks good. But could you add one more test? See below.

File: pkg/verification/sender/twilio_sender_test.go (hunk 2 above)
Comment (sbryzak) [reply pkg/verification/sender/twilio_sender_test.go:78]: Done

**Review by alexeykazakov — APPROVED** [review]
Thanks for adding the test. The tests are 70% identical so you could use sub-tests to reuse the common part (twillio configuration and gock setup) but it's minor.

**Comment by sbryzak** [comment]
I've added subtests and moved the common stuff to a reusable function.

**Pull request merged**